src/components/Tags.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react'
+import * as OF from 'office-ui-fabric-react'
+import './Tags.css'
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+    tags: string[]
+    onAdd: (tag: string) => void
+    onRemove: (tag: string) => void
+}
+
+interface State {
+    inputValue: string
+    showForm: boolean
+}
+
+class component extends React.Component<Props, State> {
+    inputRef = React.createRef<HTMLInputElement>()
+    
+    state: Readonly<State> = {
+        inputValue: '',
+        showForm: false
+    }
+
+    onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        const tag = this.state.inputValue
+        this.props.onAdd(tag)
+        this.setState({
+            inputValue: ''
+        })
+    }
+
+    onChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const inputValue = event.target.value
```

Comment:
Consider doing a toUpperCase() here so that it's always upper case

Reply:
I don't think we should modify the user's input. I think they should feel in control and we should let them do what they want. We already have this philosophy with being able to edit the dialogs.  We let them edit entities in use and invalidate things because it's what they want to do. 

The text search and comparison is still case insensitive so you can still find tags.